projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php

```diff
@@ -492,23 +491,8 @@ public static function user_can_view_post_nocache() {
 	 * @return bool
 	 */
 	public static function is_enabled_jetpack_recurring_payments() {
-		return (
-			self::is_supported_jetpack_recurring_payments() ||
-			(
-				Jetpack::is_connection_ready() &&
-				/** This filter is documented in class.jetpack-gutenberg.php */
-				! apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false )
-			)
-		);
-	}
-
-	/**
-	 * Whether the site's plan supports the Recurring Payments block.
-	 */
-	public static function is_supported_jetpack_recurring_payments() {
-		$api_available     = ( ( defined( 'IS_WPCOM' ) && IS_WPCOM ) || Jetpack::is_connection_ready() );
-		$supported_in_plan = Jetpack_Plan::supports( 'recurring-payments' );
-		return apply_filters( 'test_jetpack_is_supported_jetpack_recurring_payments', $api_available && $supported_in_plan );
+		$api_available = ( ( defined( 'IS_WPCOM' ) && IS_WPCOM ) || Jetpack::is_connection_ready() );
+		return apply_filters( 'test_jetpack_is_supported_jetpack_recurring_payments', $api_available );
```

Comment:
Maybe we can remove the filter entirely now?

Reply:
Nice catch!

Reply:
Fixed in 50ff034069f2b4f23b7b382277960bd9f3e7cf0a